src/Raven.Server/Documents/ETL/Providers/SQL/RelationalWriters/RelationalDatabaseWriter.cs

```diff
@@ -469,7 +470,16 @@ public static void SetParamValue(DbParameter colParam, SqlColumn column, List<Fu
                                 objectValue.TryGetMember(nameof(SqlDocumentTransformer.VarcharFunctionCall.Value), out object fieldValue))
                             {
                                 colParam.DbType = (DbType)Enum.Parse(typeof(DbType), dbType.ToString(), false);
-                                colParam.Value = fieldValue.ToString();
+
+                                var value = fieldValue.ToString();
+                                if (colParam.DbType == DbType.Guid && Guid.TryParse(value, out var guid) && colParam is Npgsql.NpgsqlParameter p)
```

Comment:
Why do we care this is `NpgsqlParameter ` ? Shouldn't this be broadly applicable?

Reply:
I didn't test it on other dbs, so cannot be sure, do you want to fix it here or we can open an new issue for it ?

Reply:
We need to know if this issue is specific only for PSGL or we need that for other relational dbs as well

Reply:
At glance it looks this is not only PSGL related